### ISSUE TITLE
No page refresh after cookie consent given?

### ISSUE THREAD

**Issue by ericc06** [statikbe/laravel-cookie-consent#8]
[Package version: 1.2.1]

I expected the page to be refreshed after cookie consent, but it's not the behavior I observe in my case.

Is not the page supposed to be refreshed when the user gives consent, to immediately take this setting into account?

**Comment by ericc06** [comment]
Reproduced with version 1.2.4.

**Comment by kristofser** [comment]
Hey @ericc06 
That's what step 8 & 9 are for in the documentation.

The cookie refresh handles this and an event will be fired when the cookies are updated.

**Comment by kristofser** [comment]
In the cookie-consent-js (at the bottom of the file):
```
//  Fire GTM event if dataLayer is found
    if (window.dataLayer) {
        window.dataLayer.push({event: GTM_EVENT});
    }
```

This GTM_EVENT is set in the config parameters.
This was default: 'Pageview' so when you changed to another page this would be triggered.

I switched this over to cookie_refresh. This is also the one you create in Tag Manager in steps 8 & 9.

**Comment by ericc06** [comment]
Thank you @kristofser 
A few more details on this one.

First, I checked that in my GTM account I correctly created all the elements specified in steps 8 & 9. It's OK.

I updated the package with composer.
I now have version 1.2.6.
But in the config file I still read:

`'gtm_event' => 'pageview',`

So I changed this line manually to have:

`'gtm_event' => 'cookie_refresh',`

But still no luck.
So I tried to modify the "public\vendor\cookie-consent\js\cookie-consent.js" file itself.
I replaced:

`...window.dataLayer.push({event:x})...`

with:

`...window.dataLayer.push({event:'cookie_refresh'})...`

And now it works. As soon as I click "Accept", I can see Google Analytics cookies being created without manually refreshing the page.

Do you think there can be something wrong with my configuration?

**Comment by ericc06** [comment]
[Package version: 1.2.6]

One more information @kristofser 
In "vendor\statikbe\laravel-cookie-consent\resources\js\cookie-consent.js" I can read:

```
const modalAlert = document.querySelector('.js-lcc-modal-alert');
...
const GTM_EVENT = modalAlert.dataset.gtmEvent || 'pageview';
```

And when I take a look at my website's source code:

```
<div role="dialog" aria-labelledby="lcc-modal-alert-label" aria-describedby="lcc-modal-alert-desc" aria-modal="true" class="lcc-modal lcc-modal--alert js-lcc-modal js-lcc-modal-alert" style="display: none;"
     data-cookie-key="__cookie_consent"
     data-cookie-value-analytics="2"
     data-cookie-value-marketing="3"
     data-cookie-value-both="true"
     data-cookie-value-none="false"
     data-cookie-expiration-days="365"
     data-ignored-paths=""
>
```

There is no "gtmEvent" in the dataset.

**Comment by dduupp** [comment]
Hey @ericc06 

The configurable GTM event variable was not included in the cookie modals' dataset. We've fixed this in current release 1.2.7!
